File: bindings/php/TickSettleMode.php
<?php
namespace Stencil;

final class TickSettleMode {
    const Placement = 0;
    const Quiet = 1;
    const InWorld = 2;

    public static function name(int $value): string {
        return match ($value) {
            self::Placement => 'Placement',
            self::Quiet => 'Quiet',
            self::InWorld => 'InWorld',
            default => "Unknown({$value})",
        };
    }
}

File: bindings/kotlin/src/main/kotlin/at/schem/nucleation/TickSettleMode.kt
package at.schem.nucleation

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface TickSettleModeLib: Library {
}
/** How the loaded structure is settled before tick 0.
*/
enum class TickSettleMode {
    Placement,
    Quiet,
    InWorld;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<TickSettleModeLib> = TickSettleModeLib::class.java
        internal val lib: TickSettleModeLib = Native.load("nucleation", libClass)
        fun fromNative(native: Int): TickSettleMode {
            return TickSettleMode.entries[native]
        }

        fun default(): TickSettleMode {
            return Placement
        }
    }
}
